packages/safety-suite-api/src/api/models/types.ts

```diff
@@ -944,3 +944,197 @@ export type LostRestrictedInput = {
   estimateLostDays: number;
   estimateRestrictedDays: number;
 };
+
+export type FieldType =
+  | 'boolean'
+  | 'dateTime'
+  | 'enum'
+  | 'group'
+  | 'list'
+  | 'number'
+  | 'text';
+export type RelationType = 'singleModel' | 'multiModel';
+export interface FieldTypes extends Record<FieldType, unknown> {
+  list: string;
+  number: number;
+  enum: string;
+  text: string;
+  boolean: string;
+}
+
+export interface BaseField<Type extends string> {
+  /**
+   * Field name.
+   */
+  name: Alias;
+  /**
+   * Formatted display label.
+   */
+  label?: string;
+  /**
+   * The data type.
+   */
+  type: Type;
+}
+
+export interface BaseFieldOptions<Value extends unknown> {
+  /**
+   * How the value should be formatted. E.g. `currency` for a number type.
+   */
+  formatType?: string;
+  /**
+   * Default value if no value is provided upon creation of the model.
+   */
+  defaultValue?: Value;
+  /**
+   * If the field is included in custom reporting. Can be a boolean or an object to be checked against config.
+   *
+   * Defaults to `true`
+   */
+  onReport?: boolean | {key: string; values: unknown[]};
+  onColumn?: boolean;
+  /**
+   * Indirect modules which use this field.
+   */
+  indirectModules?: Alias[];
+}
+
+export type Required =
+  | boolean
+  | {
+      path: string[];
+      cases: {
+        pattern: string;
+        result: boolean;
+      }[];
+      default?: boolean;
+    };
+
+export type DateType = 'date' | 'time' | 'dateTime';
+
+export interface Field<Type extends FieldType = FieldType>
+  extends BaseField<Type> {
+  options: BaseFieldOptions<FieldTypes[Type]> & {
+    /**
+     * If the field is confidential.
+     */
+    protected: boolean;
+    /**
+     * If the field is required, can be a boolean or an object to be checked against config.
+     */
+    required: Required;
+    sideEffect?: unknown;
```

Comment:
Similar to computations, I can map this out for you, but its BE only so don't know if its worth it.

Reply:
Yeah, my thought is that it's probably not worth the effort typing it when we'll most likely never use it on the FE.